### PR TITLE
refactor: Device data collection

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

To separate the collection of device data from its serialisation, supporting the Unity C# layer changes.

## Changeset

- Remove all methods from `DeviceData` and `DeviceDataSummary` which aren't accessors or serialisers
- Create `DeviceDataCollector`, which collects data for `DeviceData`, `DeviceDataSummary`, and device `MetaData`.
- Cache suitable values in `DeviceDataCollector` compared to `DeviceData`
- Generate a new `DeviceData` instance for each error report.
- Add `Client#populateDeviceMetaData(MetaData metaData)` method for accessing extra device data as internal API

## Tests

Ran mazerunner and existing unit test suite. Added unit tests for new methods.

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [ ] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [ ] Consistency across platforms for structures or concepts added or modified
- [ ] Consistency between the changeset and the goal stated above
- [ ] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [ ] Usage friction - is the proposed change in usage cumbersome or complicated?
- [ ] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [ ] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [ ] Thoroughness of added tests and any missing edge cases
- [ ] Idiomatic use of the language

### CODE REVIEW

File: sdk/src/main/java/com/bugsnag/android/DeviceDataSummary.java [hunk 1]
```diff
@@ -2,32 +2,28 @@
...
 
-import java.io.File;
 import java.io.IOException;
 
 public class DeviceDataSummary implements JsonStream.Streamable {
 
-    private boolean rooted = isRooted();
+    private boolean rooted;
 
+    @SuppressWarnings("NullableProblems") // set after initialisation
```
Comment (fractalwrench) [inline sdk/src/main/java/com/bugsnag/android/DeviceDataSummary.java]: explained here: https://github.com/bugsnag/bugsnag-android/pull/339/files#r196763000
